azurerm/internal/services/mssql/resource_arm_mssql_database.go

```diff
@@ -0,0 +1,363 @@
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.MsSqlDatabaseAutoPauseDelay,
+			},
+
+			//recovery is not support in version 2017-10-01-preview
+			"create_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.CreateModeCopy),
+					string(sql.CreateModeDefault),
+					string(sql.CreateModeOnlineSecondary),
+					string(sql.CreateModePointInTimeRestore),
+					string(sql.CreateModeRestore),
+					string(sql.CreateModeRestoreExternalBackup),
+					string(sql.CreateModeRestoreExternalBackupSecondary),
+					string(sql.CreateModeRestoreLongTermRetentionBackup),
+					string(sql.CreateModeSecondary),
+				}, false),
+			},
+
+			"collation": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.MsSqlDBCollation(),
+			},
+
+			"elastic_pool_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.MsSqlElasticPoolID,
+			},
+
+			"license_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.BasePrice),
+					string(sql.LicenseIncluded),
+				}, false),
+			},
+
+			"max_size_gb": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 4096),
+			},
+
+			"min_capacity": {
+				Type:         schema.TypeFloat,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: azValidate.FloatInSlice([]float64{0.5, 0.75, 1, 1.25, 1.5, 1.75, 2}),
+			},
+
+			"restore_point_in_time": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: suppress.RFC3339Time,
+				ValidateFunc:     validation.IsRFC3339Time,
+			},
+
+			"read_replica_count": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 4),
+			},
+
+			"read_scale": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.DatabaseReadScaleEnabled),
+					string(sql.DatabaseReadScaleDisabled),
+				}, false),
+			},
+
+			"sample_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.AdventureWorksLT),
+				}, false),
+			},
+
+			// hyper_scale can not be changed into other sku
+			"sku_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Computed:         true,
+				ValidateFunc:     validate.MsSqlDBSkuName(),
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"source_database_id": {
```

Comment:
Would this make more sense as
```suggestion
			"creation_source_database_id": {
```